docs/src/account/components.md

```diff
@@ -9,180 +9,201 @@ Account components are reusable units of functionality that define a part of an
-    { value = "0x0" }
+[[storage.value]]
+name = "demo::token_metadata"
+description = "Contains token metadata (max supply, symbol, decimals)."
+type = [
+    { type = "u32", name = "max_supply", description = "Maximum supply of the token in base units" },
+    { type = "token_symbol", name = "symbol", description = "Token symbol", default-value = "TST" },
+    { type = "u8", name = "decimals", description = "Number of decimal places for converting to absolute units" },
+    { type = "void" }
 ]
 
-[[storage]]
-name = "owner_public_key"
-description = "This is a value placeholder that will be interpreted as a Falcon public key"
-slot = 1
+[[storage.value]]
+name = "demo::owner_public_key"
+description = "This is a typed value supplied at instantiation and interpreted as a Falcon public key"
 type = "auth::rpo_falcon512::pub_key"
 
-[[storage]]
-name = "map_storage_entry"
-slot = 2
-values = [
-    { key = "0x1", value = ["0x0", "249381274", "998123581", "124991023478"] },
-    {
-      key = "0xDE0B1140012A9FD912F18AD9EC85E40F4CB697AE",
-      value = {
-        name = "value_placeholder",
-        description = "This value will be defined at the moment of instantiation"
-      }
-    }
+[[storage.value]]
+name = "demo::protocol_version"
+description = "A whole-word init-supplied value typed as a felt (stored as [0,0,0,<value>])."
+type = "u8"
+
+[[storage.map]]
+name = "demo::static_map"
+description = "A map slot with statically defined entries"
+default-values = [
+    { key = "0x0000000000000000000000000000000000000000000000000000000000000001", value = ["0x0", "249381274", "998123581", "124991023478"] },
+    { key = ["0", "0", "0", "2"], value = "0x0000000000000000000000000000000000000000000000000000000000000010" }
 ]
 
-[[storage]]
-name = "procedure_thresholds"
+[[storage.map]]
+name = "demo::procedure_thresholds"
 description = "Map which stores procedure thresholds (PROC_ROOT -> signature threshold)"
-slot = 3
-type = "map"
-
-[[storage]]
-name = "multislot_entry"
-slots = [4,5]
-values = [
-    ["0x1","0x2","0x3","0x4"],
-    ["50000","60000","70000","80000"]
-]
+key-type = "word"
+value-type = "u16"
 ```
 
-#### Specifying values and their types
+#### Specifying word schema and types
+
+Value-slot entries describe their schema via `WordSchema`. A value type can be either:
 
-In the TOML format, any value that is one word long can be written as a single value, or as exactly four field elements. In turn, a field element is a number within Miden's finite field.
+- **Singular**: defined through the `type` field, indicating the expected `SchemaTypeIdentifier` for the entire word. The value is supplied at instantiation time via `InitStorageData`.
+- **Composed**: provided through `type = [ ... ]`, which contains exactly four `FeltSchema` descriptors. Each element is either a named typed field (optionally with `default-value`) or a `void` element for reserved/padding zeros.
```

Comment:
As mentioned in another comment, I'd replace "Composed" with "Composite". Also, maybe "Singular" could be "Simple" - but I'm not too sure about this.